### ISSUE TITLE
Various fixes

### ISSUE THREAD

**Issue by PiKeyAr** [SonicFreak94/sadx-fixed-edition#2]
Eggman NPC model

```c++
*(NJS_OBJECT *)0x010FEF74 = *(NJS_OBJECT *)0x02EEB524; //Replace the non-updated Eggmobile NPC model with a high-poly one to resolve a texture issue

WriteData((NJS_TEXLIST**)0x007D2B22, (NJS_TEXLIST*)0x02EE0AA4); //Replace the texlist for the above model in the NPC data array
```

```c++
*(NJS_TEXLIST**)0x02BD5FE4 = (NJS_TEXLIST*)0x02EE0AA4; //Texlist fix for Eggman in Super Sonic story
```

E101 Beta lighting fix
```c++
((NJS_OBJECT*)0x014D857C)->basicdxmodel->mats[1].attrflags |= NJD_FLAG_IGNORE_LIGHT;
((NJS_OBJECT*)0x014D857C)->basicdxmodel->mats[2].attrflags |= NJD_FLAG_IGNORE_LIGHT;
((NJS_OBJECT*)0x014D887C)->basicdxmodel->mats[1].attrflags |= NJD_FLAG_IGNORE_LIGHT;
((NJS_OBJECT*)0x014D943C)->basicdxmodel->mats[7].attrflags |= NJD_FLAG_IGNORE_LIGHT;
((NJS_OBJECT*)0x014DC25C)->basicdxmodel->mats[0].attrflags |= NJD_FLAG_IGNORE_LIGHT;
((NJS_OBJECT*)0x014DD4A4)->basicdxmodel->mats[0].attrflags |= NJD_FLAG_IGNORE_LIGHT;
```

Zero w/Amy
```c++
((NJS_OBJECT *)0x31A4DFC)->basicdxmodel->mats[11].attrflags &= ~NJD_FLAG_IGNORE_LIGHT; //Zero holding Amy lighting fix
```

Eggman model lighting fix
```c++
((NJS_OBJECT*)0x008961E0)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x008964CC)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x008980DC)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x00897DE0)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
```

Eggman model (Eggmobile) lighting fix
```c++
((NJS_OBJECT*)0x02EE22C0)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x02EE25AC)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x02EE4194)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x02EE3E98)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
```

Tikal lighting fixes
```c++
((NJS_OBJECT*)0x008CE058)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
((NJS_OBJECT*)0x008CC658)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
```

Twinkle Park barrel model (requires a different "boo" texture in OBJ_TWINKLE!)
```c++
*(NJS_OBJECT*)0x0279D364 = object_000A0E58;
```
(sorted SA1 model attached)
[barrel.txt](https://github.com/SonicFreak94/sadx-fixed-edition/files/818925/barrel.txt)

**Comment by PiKeyAr** [comment]
I suggest adding this to the Eggmobile fix:
((NJS_OBJECT*)0x02EEB524)->basicdxmodel->mats[0].attrflags &= ~NJD_FLAG_USE_ALPHA;
Both "fixed" and "wrong" models have the alpha flag enabled on one of the materials, which cases visual weirdness like this:
![image](https://user-images.githubusercontent.com/23312549/29155843-6f650cb6-7dd9-11e7-9318-e9dc3ffe6a82.png)
The alpha flag isn't needed anyway because the texture used on that material is opaque and the alpha value is 255 anyway.

**Comment by PiKeyAr** [comment]
I recently found out that replacing the Eggmobile NPC model with the higher-poly one causes the following issue:
![Uploading 3XUP-KXO.jpg large.jpg…]()
It's caused by several material flags that indicate that the material should use alpha, even though the textures are opaque and the alpha value is 255. Disabling the alpha flag fixes the issue, but causes another problem when Eggman's body can't be seen through the front glass.
I've redone my fix by editing the "wrong" NPC model instead of replacing it with the higher-poly model.
Fixed model attached.
The following line needs to be added to init instead of the previous fix:
*(NJS_OBJECT*)0x010FEF74 = object_00CFEF74; //Fix materials on Eggmobile NPC model
[EggmobileNPC.txt](https://github.com/SonicFreak94/sadx-fixed-edition/files/1215043/EggmobileNPC.txt)
